docker/client.py

```diff
@@ -140,9 +140,14 @@ def build(self, path=None, tag=None, quiet=False, fileobj=None,
             if self._auth_configs:
                 if headers is None:
                     headers = {}
-                headers['X-Registry-Config'] = auth.encode_full_header(
-                    self._auth_configs
-                )
+                if utils.compare_version('1.19', self._version) >= 0:
```

Comment:
The only issue that this causes, for my particular situation, is that `docker-compose` would still be broke since it registers the [client API version as 1.18](https://github.com/docker/compose/blob/master/compose/cli/docker_client.py#L35). I don't know really how to detect what version the docker server is expecting. I'm not sure that an API version was changed when this authentication header was changed.